libs/application/templates/children-residence-change/src/dataProviders/MockNationalRegistryProvider.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Application,
+  BasicDataProvider,
+  FailedDataProviderResult,
+  SuccessfulDataProviderResult,
+} from '@island.is/application/core'
+
+import { PersonResidenceChange } from '@island.is/application/templates/children-residence-change'
+import { NationalRegistry, DataProviderTypes, CRCApplication } from '../types'
+
+export class MockNationalRegistryProvider extends BasicDataProvider {
+  readonly type = DataProviderTypes.MOCK_NationalRegistry
+
+  async provide(application: Application): Promise<NationalRegistry> {
+    const query = `
+      query NationalRegistryUserQuery {
+        nationalRegistryUser {
+          nationalId
+          fullName
+          address {
+            code
+            postalCode
+            city
+            streetAddress
+            lastUpdated
+          }
+        }
+      }
+    `
+    const crcApplication = (application as unknown) as CRCApplication
```

Comment:
Do want to use typed props and the `Override` function from `libs/application/templates/children-residence-change/src/types/index.ts` instead of throwing to unknown?

Reply:
I can't figure out how to make this work 🤷